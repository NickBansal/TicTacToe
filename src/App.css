.App {
  text-align: center;
}

#ResetModal {
  max-height: calc(100% - 100px);
  position: fixed;
  top: 32%;
  left: 50%;
  transform: translate(-50%, -50%);
  width: 490px;
  height: 230px;
  background: rgba(0, 0, 0, 0.9);
  animation: modalFade 0.5s 1 ease-in-out
}

.Winner {
  color: whitesmoke
}

@keyframes modalFade {
  from {
    opacity: 0
  }
  to {
    opacity: 1;
  }
}


