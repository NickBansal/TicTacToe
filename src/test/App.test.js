import React from 'react';
import ReactDOM from 'react-dom';
import App from '../TicTacToe';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json'

configure({ adapter: new Adapter() });

describe.only('<App />', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(<App />);
  })
  it('includes 1 div with the className of App', () => {
    expect(wrapper.find('div.App').exists()).toBe(true)
  })
  it('includes 2 divs', () => {
    expect(wrapper.find('div.App h1').text()).toBe("Tic Tac Toe")
  })
  it('Component matches the snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  it('Testing the state components', () => {
    expect(wrapper.state().squares).toEqual([null, null, null, null, null, null, null, null, null]);
    expect(wrapper.state().difficulty).toBe('Easy')
  })
})
