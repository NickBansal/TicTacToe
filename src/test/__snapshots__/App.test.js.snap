// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Component matches the snapshot 1`] = `
<div
  className="App"
>
  <div
    style={null}
  >
    <h1>
      Tic Tac Toe
    </h1>
    <h3
      style={
        Object {
          "margin": "20px",
        }
      }
    >
      Difficulty: 
      Easy
    </h3>
    <Board
      gameSquares={
        Array [
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null,
        ]
      }
      handleClick={[Function]}
    />
    <button
      onClick={[Function]}
      value="Easy"
    >
      EASY
    </button>
    <button
      onClick={[Function]}
      value="Hard"
    >
      HARD
    </button>
  </div>
</div>
`;
